File: bank-service/src/main/java/rs/edu/raf/dto/CreditRequestDTO.java
package rs.edu.raf.dto;

import java.math.BigDecimal;

public record CreditRequestDTO(Long id, CreditTypeDTO creditType, BigDecimal loanAmount, CurrencyDTO currency,
                               String loanPurpose, BigDecimal salary, String phoneNumber, boolean permanentEmployee,
                               int currentEmploymentPeriod, int loanTerm, BankAccountDTO bankAccount,
                               String creditRequestStatus) {
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/dto/CreditRequestDTO.kt
package org.example.bankservicekotlin.dto

import java.math.BigDecimal

data class CreditRequestDTO(
    val id: Long?,
    val creditType: CreditTypeDTO,
    val loanAmount: BigDecimal,
    val loanPurpose: String,
    val salary: BigDecimal,
    val phoneNumber: String,
    val permanentEmployee: Boolean,
    val currentEmploymentPeriod: Int,
    val loanTerm: Long,
    val bankAccount: BankAccountDTO,
    val creditRequestStatus: String
)
